src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -285,5 +294,53 @@ private void AddMetadataFromCustomProviders(IEnumerable<IFunctionProvider> funct
                 }
             }
         }
+
+        private void HostJsonFileValidator()
+        {
+            try
+            {
+                // Search for the host.json file within nested directories to verify scenarios where it isn't located at the root. This situation often occurs when a function app has been improperly zipped.
+                string hostFilePath = Path.Combine(_scriptOptions.Value.RootScriptPath, ScriptConstants.HostMetadataFileName);
+                IEnumerable<string> hostJsonFiles = Directory.GetFiles(_scriptOptions.Value.RootScriptPath, ScriptConstants.HostMetadataFileName, SearchOption.AllDirectories)
+                    .Where(file => !file.Equals(hostFilePath, StringComparison.OrdinalIgnoreCase));
+
+                if (IsDefaultHostConfig())
+                {
+                    if (hostJsonFiles != null && hostJsonFiles.Any())
+                    {
+                        string hostJsonFilesPath = string.Join(", ", hostJsonFiles).Replace(_scriptOptions.Value.RootScriptPath, string.Empty);
+                        _logger.HostJsonZipDeploymentIssue(hostJsonFilesPath);
+                    }
+                    else
+                    {
+                        _logger.NoHostJsonFile();
+                    }
+                }
+                else
+                {
+                    _logger.MissingFunctionsDeploymentIssue();
```

Comment:
If cx app has host.json file at the root folder (and Cx has added field `isDefaultHostConfig` to true) but the app does not have any functions. We will not be printing "no functions found" message for that case.

To avoid customer using this variable incorrectly, we need to have its correct usage in public documentation.

Reply:
we should add tests to make sure it prints "no functions found" for the valid cases

Reply:
@RohitRanjanMS is this the log we discussed about removing?

Reply:
@surgupta-msft, yes this is the log. I have updated the PR. Removed log for this scenario.

Reply:
We have 2 scenarios now - 
1. host.json doesn't exist at the root (or the customer added a host.json with the `isDefaultHostConfig` flag), and there is a host.json in one of the subfolders of the zip package, resulting in 0 functions found.
```
Error - No functions were found. A valid host.json file wasn't found in the package root. However, one was located at: {hostJsonFilesPath}. This state indicates that your deployment package was created incorrectly. For deployment package requirements, see https://aka.ms/deployment-zip-push.
```
2. host.json doesn't exist at the root (or the customer added a host.json with the `isDefaultHostConfig` flag) and 0 functions found. 
```
Information - No functions were found. This can occur before you deploy code to your function app or when the host.json file is missing from the most recent deployment. Make sure that your deployment package includes the host.json file in the root of the package. For deployment package requirements, see https://aka.ms/functions-deployment-technologies.
```

We discussed the misuse of isDefaultHostConfig and how it might influence this logic, but chose not to document it to avoid confusion. `isDefaultHostConfig` in host.json won't have any impact unless we encounter the 0 functions found issue.